### PR TITLE
Add --json option to get balance as JSON

### PR DESCRIPTION
**Pull request by sculptex**
usd - Current USD value,
zcn - ZCN amount (float)
fmt - Formatted zcn value (mZCN etc.)

A brief description of the changes in this PR:

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zwalletcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- 0chain:
- Other: ...

### CODE REVIEW

**Comment by sculptex** [comment]
Actually, for true JSON support for such commands, we could have a success variable. Only when success:true would the user then read the rest of the values. An alternative would be just to return relevant parameters of ZCN being 0.

For now, I will just class invalid JSON as 0.

File: cmd/getbalance.go [hunk 1]
```diff
@@ -14,6 +15,8 @@ var getbalancecmd = &cobra.Command{
 	Long:  `Get balance from sharders`,
 	Args:  cobra.MinimumNArgs(0),
 	Run: func(cmd *cobra.Command, args []string) {
+		doJSON, _ := cmd.Flags().GetBool("json")
```
Comment (peterlimg) [inline cmd/getbalance.go:18]: Check error in case use invalid value is used

File: cmd/getbalance.go [hunk 2]
```diff
@@ -29,6 +32,15 @@ var getbalancecmd = &cobra.Command{
 		}
 		b := statusBar.balance
 		usd, err := zcncore.ConvertTokenToUSD(b.ToToken())
+		
+		if doJSON {
+			j := make(map[string]string)
+			j["usd"] = fmt.Sprintf("%f", usd)
+			j["zcn"] = fmt.Sprintf("%f", b.ToToken())
+			j["fmt"] = fmt.Sprintf("%s", b)
+			util.PrintJSON(j)
```
Comment (peterlimg) [inline cmd/getbalance.go:41]: We can it in this way:
```go
j := map[string]string {
  "usd": fmt.Sprintf("%f", usd),
  "zcn": fmt.Sprintf("%f", b.ToToken())
...
}
```

**Review by peterlimg — DISMISSED** [review]

File: cmd/getbalance.go (hunk 1 above)
Comment (sculptex) [reply cmd/getbalance.go:18]: --json is bool so requires no parameter. But feel free to change however is best way

File: cmd/getbalance.go (hunk 2 above)
Comment (sculptex) [reply cmd/getbalance.go:41]: Sure I see this is more compact, please implement as you see fit

**Comment by sculptex** [comment]
@peterlimg what is your opinion on having success field?
So that with --json, valid json is always returned, even if just {"success"=false} or whatever.
But then client only checks other values if success=true?

**Comment by peterlimg** [comment]
> @peterlimg what is your opinion on having success field? So that with --json, valid json is always returned, even if just {"success"=false} or whatever. But then client only checks other values if success=true?

I would suggest to not add the success field, we can return json value when success, and return error directly. We can put the command in `RunE()`, so when error happens, it will have a formatted message returned, i.e, "Error: xxxx", so if the CLI is used by other services, the prefix of "Error" could be checked.

**Comment by sculptex** [comment]
Okay, sure.

**Comment by sculptex** [comment]
@peterlimg sorry, I'm not too hot with GitHub, am I required to implement your changes before you can approve and close?

File: cmd/getbalance.go (hunk 1 above)
Comment (sculptex) [reply cmd/getbalance.go:18]: @peterlimg do you agree? there is no parameter to check?

File: cmd/getbalance.go (hunk 2 above)
Comment (sculptex) [reply cmd/getbalance.go:41]: Okay, I implemented this

**Comment by sculptex** [comment]
@peterlimg can this be merged now?

**Review by cnlangzi — APPROVED** [review]
LGTM

**Comment by sculptex** [comment]
why wasn't this merged? now its conflicting again..

**Pull request merged**